### ISSUE TITLE
Plugin Tracking Issue in Ionic 4

### ISSUE THREAD

**Issue by Buddies2019** [transistorsoft/cordova-background-geolocation-lt#974]
## Your Environment
* Plugin version:
* Platform: iOS or Android
* OS version:
* Device manufacturer / model:
* Cordova version (`cordova -v`):
* Cordova platform version (`cordova platform ls`):
* Plugin config provided to `#ready`:
```javascript <!-- syntax-highlighting:  paste your code below -->
INSERT_YOUR_CODE_HERE
```

## Steps to Reproduce

1.
2.
3.
4.

## Debug logs

<details>
	<summary>Logs</summary>

```<!-- Syntax highlighting:  DO NOT REMOVE -->
PASTE_YOUR_LOGS_HERE
```

</details>

**Comment by Buddies2019** [comment]
Hi Chris ,

kindly find the below config which we use your plugin ,  Plugin didnt send any Lat , lag some times  i hope service can stop how to rewake and let me know if any in our Config File ,

we need only GPS Location not for Cellular because some times Geo location get from Tower .

const configuration = {
      reset: false /  Set to false for Optimization /,
      debug: true,
      logLevel: BackgroundGeolocation.LOG_LEVEL_VERBOSE,
      desiredAccuracy: BackgroundGeolocation.DESIRED_ACCURACY_HIGH,
      distanceFilter: 10,
      url: url + "/cars/demo_post?id=" + this.credentials.id,
      autoSync: true,
      /* autoSyncThreshold: 5,
       batchSync: false,*/
      stopOnTerminate: false,
      startOnBoot: true,
      stopOnStationary: false,
      activityType: BackgroundGeolocation.ACTIVITY_TYPE_OTHER_NAVIGATION
    };
    // 2.  Configure the plugin with #ready
    BackgroundGeolocation.ready(configuration, (state) => {
      // console.log('[ready] BackgroundGeolocation is ready to use');
      if (state.enabled === false) {
        // 3.  Start tracking.
        BackgroundGeolocation.start();

      }
});

**Comment by Buddies2019** [comment]
{
  "activityRecognitionInterval": 10000,
  "allowIdenticalLocations": false,
  "autoSync": true,
  "autoSyncThreshold": 0,
  "batchSync": false,
  "debug": true,
  "deferTime": 0,
  "desiredAccuracy": 0,
  "desiredOdometerAccuracy": 100,
  "disableElasticity": false,
  "disableLocationAuthorizationAlert": false,
  "disableStopDetection": false,
  "distanceFilter": 10,
  "elasticityMultiplier": 1,
  "enableHeadless": true,
  "enableTimestampMeta": false,
  "extras": {},
  "fastestLocationUpdateInterval": -1,
  "forceReloadOnBoot": false,
  "forceReloadOnGeofence": false,
  "forceReloadOnHeartbeat": false,
  "forceReloadOnLocationChange": false,
  "forceReloadOnMotionChange": false,
  "forceReloadOnSchedule": false,
  "foregroundService": true,
  "geofenceInitialTriggerEntry": true,
  "geofenceModeHighAccuracy": false,
  "geofenceProximityRadius": 1000,
  "geofenceTemplate": "",
  "headers": {},
  "headlessJobService": "com.transistorsoft.cordova.bggeo.BackgroundGeolocationHeadlessTask",
  "heartbeatInterval": -1,
  "httpRootProperty": "location",
  "httpTimeout": 60000,
  "isMoving": false,
  "locationTemplate": "",
  "locationTimeout": 60,
  "locationUpdateInterval": 1000,
  "locationsOrderDirection": "ASC",
  "logLevel": 5,
  "logMaxDays": 3,
  "maxBatchSize": -1,
  "maxDaysToPersist": 14,
  "maxRecordsToPersist": -1,
  "method": "POST",
  "minimumActivityRecognitionConfidence": 75,
  "notification": {
    "layout": "",
    "title": "Activity Sync",
    "text": "Activity Tracking Engaged",
    "color": "",
    "channelName": "TSLocationManager",
    "smallIcon": "",
    "largeIcon": "",
    "priority": 0,
    "strings": {},
    "actions": []
  },
  "params": {
    "device": {
      "model": "LLD-AL10",
      "platform": "Android",
      "uuid": "LLD-AL10",
      "version": "9",
      "manufacturer": "HUAWEI",
      "framework": "Cordova"
    }
  },
  "persist": true,
  "persistMode": 2,
  "schedule": [],
  "scheduleUseAlarmManager": false,
  "speedJumpFilter": 300,
  "startOnBoot": true,
  "stationaryRadius": 25,
  "stopAfterElapsedMinutes": 0,
  "stopOnStationary": false,
  "stopOnTerminate": false,
  "stopTimeout": 1,
  "triggerActivities": "in_vehicle, on_bicycle, on_foot, running, walking",
  "url": "http:\/\/api.b2taxi.in:55577\/cars\/demo_post?id=2",
  "useSignificantChangesOnly": false,
  "enabled": true,
  "schedulerEnabled": false,
  "trackingMode": 1,
  "odometer": 0,
  "isFirstBoot": false
}

**Comment by christocracy** [comment]
Do not ignore the issue template:

-Plugin version:
-Platform: iOS or Android
-OS version:
-Device manufacturer / model:
-Cordova version (cordova -v):
-Cordova platform version (cordova platform ls):

**Comment by Buddies2019** [comment]
i dont have a clue what is the exact Issue Past few days i am trying to fix, i have launch app within a week kindly help me here

**Comment by christocracy** [comment]
Answer the following questions (which you ignored in the issue template):

- Plugin version?:
- Platform: iOS or Android?
- OS version?:
- Device manufacturer / model:?
- Cordova version (cordova -v):?
- Cordova platform version (cordova platform ls):?

**Comment by Buddies2019** [comment]
Plugin version:3.0.8
Platform: iOS or Android : Android
OS version: Android 9
Device manufacturer / model:HUAWEI LLD-AL10 
cordova (Cordova CLI) : 8.1.2 (cordova-lib@8.1.1)
   Cordova Platforms     : android 8.0.0, browser 5.0.4
   Cordova Plugins       : cordova-plugin-ionic 5.4.3, cordova-plugin-ionic-keyboard 2.1.3, cordova-plugin-ionic-webview 4.1.1, (and 33 other plugins)

**Comment by christocracy** [comment]
Huawei is the worst possible device for app that operate in the background.  See http://dontkillmyapp.com

**Comment by Buddies2019** [comment]
all the Device not only for Huawei 

<img width="514" alt="Capture" src="https://user-images.githubusercontent.com/53367881/62472361-242a5700-b7bc-11e9-91b7-3e1adc03b1f5.PNG">
)

**Comment by Buddies2019** [comment]
when plugin went stationary Mode after didnt wakeup until application open , how to avoid cellular location because cellular give wrong lat,lag

**Comment by christocracy** [comment]
Build and install the [SampleApp](https://github.com/transistorsoft/cordova-background-geolocation-SampleApp) on your device.  

>how to avoid cellular location because cellular give wrong lat,lag

You can't.  Device location-services returns the *best available* location.  If GPS is available (ie: when you outdoors), the location will automatically come from GPS.  Otherwise, location *automatically* comes from Wifi or Cellular.

**Comment by Buddies2019** [comment]
yes we can try ,how we can get Only GPS Location , how we ignore Cellular Location ?

**Comment by christocracy** [comment]
>how we ignore Cellular Location ?

You can't.

**Comment by Buddies2019** [comment]
then wrong Km will comes , how we can fix

**Comment by christocracy** [comment]
If location came from Cellular (typical accuracy: 1000), those location won't be used to update odometer.  See [Config.desiredOdometerAccuracy](https://transistorsoft.github.io/cordova-background-geolocation-lt/interfaces/_cordova_background_geolocation_lt_.config.html#desiredodometeraccuracy).

**Comment by Buddies2019** [comment]
thanks i will check the same

**Comment by Buddies2019** [comment]
hi now working fine there is one concen , if we travel than also so its showing is_moving false for few meter some time km also is this case we have close app and reopen then it will start , please advice we can do manually  start while is stop while Mobile Moving ?

**Comment by christocracy** [comment]
See docs `BackgroundGeolocation.changePace`

**Comment by rampujari** [comment]
Hi, I want to get location data when i kill application after that every 5 seconds or 10 seconds is this possible on ios?

Thank you in advances.

**Comment by christocracy** [comment]
@rampujari Your question has nothing to do with this issue, which is related to *Android*.
